api/controllers/v0.2/createPatron.js

```diff
@@ -55,74 +55,155 @@ function renderResponse(req, res, status, message) {
+    process.env.PATRON_STREAM_NAME_V02,
+    streamPatronData // eslint-disable-line comma-dangle
+  ).then((streamResponse) => { // eslint-disable-line no-unused-vars
+    renderResponse(req, res, 201, modeledResponse);
+    console.log('Published to stream successfully!'); // eslint-disable-line no-console
+  }).catch((streamError) => {
+    renderResponse(req, res, 201, streamError);
+    console.error(`Error publishing to stream: ${streamError}`); // eslint-disable-line no-console
+  });
+}
 
-    return;
-  }
+/**
+ * callAxiosToCreatePatron(req, res, tokenResponse)
+ * Stream the new patron data to Kinesis
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ * @param {object} tokenResponse
+ */
+function callAxiosToCreatePatron(req, res, tokenResponse) {
+  axios.post(process.env.ILS_CREATE_PATRON_URL, req.body, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${tokenResponse.data.access_token}`,
+    },
+  })
+    .then((response) => {
+      const modeledResponse =
+        modelResponse.patronCreator(response.data, response.status, req.body);
+      modelStreamPatron.transformPatronRequest(
+        req.body, modeledResponse // eslint-disable-line comma-dangle
+      )
+        .then((streamPatronData) => {
+          streamPatron(req, res, streamPatronData, modeledResponse);
+        })
+        .catch((error) => {
+          renderResponse(req, res, 201, modeledResponse);
+          console.error(`Error creating patron: ${error}`); // eslint-disable-line no-console
+        });
+    })
+    .catch((response) => {
+      // eslint-disable-next-line no-console
+      console.error(
+        `status_code: ${response.response.status}, ` +
+        'type: "invalid-request", ' +
+        `message: "${response.message} from ILS.", ` +
+        `response: ${JSON.stringify(response.response.data)}` // eslint-disable-line comma-dangle
+      );
+
+      if (response.response && response.response.data) {
+        const responseObject = collectErrorResponseData(
+          response.response.status,
+          response.response.data.type,
+          response.response.data.detail,
+          response.response.data.title,
+          response.response.data.debug_message // eslint-disable-line comma-dangle
+        );
+
+        const statusCode = (responseObject.status) ? responseObject.status : 500;
+
+        renderResponse(
+          req,
+          res,
+          statusCode,
+          modelResponse.errorResponseData(responseObject) // eslint-disable-line comma-dangle
+        );
+      } else {
+        renderResponse(req, res, 500, modelResponse.errorResponseData(
+          collectErrorResponseData(null, '', '', '', '') // eslint-disable-line comma-dangle
+        ));
+      }
+    });
+}
 
-  if (!process.env.ILS_CREATE_PATRON_URL || isEmpty(process.env.ILS_CREATE_PATRON_URL)) {
+/**
+ * validateEnvVariable(envVariableName)
+ * Validate that a specific environment variable is present.
```

Comment:
I think this needs more explanation. *How* does it validate a specific env var? I would expect it to throw an error if the env var is missing/invalid - or just return `false` - but this seems to write directly to the HTTPResponse object (implying that no further processing is necessary/expected). It doesn't look like the caller is checking to see if the env var validated. In effect, I think this function may call `res.status(500)...` for each missing env variable in series.